### PR TITLE
GraphQL DX: Relation/Pointer

### PR DESCRIPTION
**Pull request by Moumouls**
#5863

* Change DX for Relation
* Change DX for Pointer
* Add a deep GraphQL Query test

### Relation

```graphql
input RelationInput {
   objectId: ID!
}

input ExampleRelationInput {
  add: [RelationInput!]
  addAndCreate: [CreateExampleFieldsInput!]
  remove: [RelationInput!]
}
```

### Pointer

```graphql
input PointerInput {
   objectId: ID!
}

# You need to choose link OR addAndLink
input ExamplePointerInput {
  link: PointerInput
  addAndLink: [CreateExampleFieldsInput!]
}
```

### CODE REVIEW

**Comment by Moumouls** [comment]
## `createAndAdd` and `createAndLink` support deep nested creation ✊

> Example

```js
const {
  data: { createCountry: result },
} = await apolloClient.mutate({
  mutation: gql`
    mutation CreateCountry($fields: CreateCountryFieldsInput) {
      createCountry(fields: $fields) {
        objectId
        name
        companies {
          results {
            objectId
            name
            teams {
              results {
                objectId
                name
              }
            }
          }
        }
      }
    }
  `,
  variables: {
    fields: {
      name: 'imACountry2',
      companies: {
        createAndAdd: [
          {
            name: 'imACompany2',
            teams: {
              createAndAdd: {
                name: 'imATeam2',
              },
            },
          },
          {
            name: 'imACompany3',
            teams: {
              createAndAdd: {
                name: 'imATeam3',
              },
            },
          },
        ],
      },
    },
  },
});
```

**Comment by Moumouls** [comment]
@davimacedo ready for review 👍

File: src/GraphQL/loaders/parseClassMutations.js [hunk 1]
```diff
@@ -110,10 +72,20 @@ const load = function(
...
-          transformTypes('create', fields);
+          let parseFields;
+          try {
+            parseFields = await transformTypes('create', fields, {
+              className,
+              parseGraphQLSchema,
+              req: { config, auth, info },
+            });
+          } catch (e) {
+            console.log(e);
```
Comment (davimacedo) [inline src/GraphQL/loaders/parseClassMutations.js]: Can you use the logger instead?

File: src/GraphQL/loaders/parseClassMutations.js [hunk 2]
```diff
@@ -171,13 +143,22 @@ const load = function(
...
-          transformTypes('update', fields);
+          let parseFields;
+          try {
+            parseFields = await transformTypes('update', fields, {
+              className,
+              parseGraphQLSchema,
+              req: { config, auth, info },
+            });
+          } catch (e) {
+            console.log(e);
```
Comment (davimacedo) [inline src/GraphQL/loaders/parseClassMutations.js]: Same thing here?

**Comment by douglasmuraoka** [comment]
@Moumouls I've tried some operations, such as updating an object removing one element from the relation and linking a pointer, and I got the following error (although the operation worked):

`Cannot return null for non-nullable field Foo.objectId.`

Query:
```
mutation {
  updateFoo(objectId:"hShcOgTmIF" fields: {
    pointerrrr: {
      link: {
        objectId: "NCc4wLTr50"
      }
    }
  }) {
    objectId
  }
}
```

Could you please take a look?

**Comment by Moumouls** [comment]
> @Moumouls I've tried some operations, such as updating an object removing one element from the relation and linking a pointer, and I got the following error (although the operation worked):
> 
> `Cannot return null for non-nullable field Foo.objectId.`
> 
> Query:
> 
> ```
> mutation {
>   updateFoo(objectId:"hShcOgTmIF" fields: {
>     pointerrrr: {
>       link: {
>         objectId: "NCc4wLTr50"
>       }
>     }
>   }) {
>     objectId
>   }
> }
> ```
> 
> Could you please take a look?

Strange, the test for `should support pointer on update` currently works
What is your GraphQL sequence ?

**Comment by douglasmuraoka** [comment]
The test `should support pointer on update` works because it's fetching more than the `objectId` of the updated object. I've changed the mutation to the above and it failed:

```
mutation Update($objectId: ID! $fields: UpdateCountryFieldsInput)  {
    updateCountry(objectId: $objectId, fields: $fields) {
        objectId
    }
}
```

**Comment by Moumouls** [comment]
Fixed, the issue was introduced in #5893 
Thanks for the review @douglasmuraoka !

**Pull request merged**